includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -3241,4 +3270,13 @@ function ( $a, $b ) {
 
 		return $included_count;
 	}
+
+	/**
+	 * Gets the @viewport CSS rules that were extracted from stylesheet, and vendor-prefixed rules like @-ms-viewport.
+	 *
+	 * @return array The rules that were extracted.
+	 */
+	public static function get_extracted_viewport_rules() {
+		return self::$extracted_viewport_rules;
+	}
```

Comment:
I don't think this should be added. Instead of exposing the extracted viewport rules, let's instead inside of `AMP_Style_Sanitizer::process_style_element()` and `AMP_Style_Sanitizer::process_link_element()` obtain the previously-obtained 'viewport_rules` inject them as a new `meta` tag into the document:

```diff
diff --git a/includes/sanitizers/class-amp-style-sanitizer.php b/includes/sanitizers/class-amp-style-sanitizer.php
index 81bfc64f0..5afe1e5a9 100644
--- a/includes/sanitizers/class-amp-style-sanitizer.php
+++ b/includes/sanitizers/class-amp-style-sanitizer.php
@@ -1315,6 +1315,26 @@ class AMP_Style_Sanitizer extends AMP_Base_Sanitizer {
 			]
 		);
 
+		if ( ! empty( $parsed['viewport_rules'] ) ) {
+			$viewport_meta = $this->dom->createElement( 'meta' );
+			$viewport_meta->setAttribute( 'name', 'viewport' );
+			$viewport_meta->setAttribute(
+				'content',
+				implode(
+					',',
+					array_map(
+						static function ( $property_name ) use ( $parsed ) {
+							return $property_name . '=' . $parsed['viewport_rules'][ $property_name ];
+						},
+						array_keys( $parsed['viewport_rules'] )
+					)
+				)
+			);
+
+			// Inject aa potential duplicate meta viewport element, to later be merged in AMP_Meta_Sanitizer.
+			$element->parentNode->insertBefore( $viewport_meta, $element );
+		}
+
 		$this->pending_stylesheets[] = [
 			'group'              => self::STYLE_AMP_CUSTOM_GROUP_INDEX,
 			'original_size'      => strlen( $stylesheet ),
```

With this in place, note that `AMP_Meta_Sanitizer` already is forced to run _after_ `AMP_Style_Sanitizer`, so what that means is that the `AMP_Style_Sanitizer` can then be updated to iterate over all `meta[name=viewport]` elements and _merge_ them together into one.